.github/scripts/trymerge.py

```diff
@@ -1511,6 +1548,12 @@ def handle_exception(e: Exception, title: str = "Merge failed") -> None:
         gh_post_pr_comment(org, project, args.pr_num, "Cross-repo ghstack merges are not supported", dry_run=args.dry_run)
         return
 
+    if not has_required_labels(pr):
+        add_label_err_comment(pr)
+        return
+    else:
+        delete_all_label_err_comments(pr)
```

Comment:
Within trymerge, the failure here isn't as loud as we'd like it to be. The `handle_exception` method formats it in a better way and makes the message easier to parse.

Can you please move this checking/commenting to be inside the `merge` method and to use the error formatting over there?

Reply:
Ah good point! Moved into merge to take advantage of existing error handling.